src/pages/TaskRun/Dependencies-Tile.vue

```diff
@@ -47,50 +84,104 @@ export default {
+            ${taskRunFields}
+          }
+        `
+            return string
+          },
+          ''
+        )
+        queryString += downstream
+      }
+
+      return queryString == ''
+        ? null
+        : gql`
+        query TaskRunDependencies {
+          ${queryString}
+        }
+      `
+    },
+    modifiedTaskRun() {
+      return {
+        ...this.taskRun,
+        flow_run_name: this.taskRun.flow_run.name,
+        task: {
+          ...this.taskRun.task,
+          task_run_id: this.taskRun.id,
+          name:
+            this.taskRun.name ??
+            this.taskRun.task.name +
+              (this.taskRun.map_index > -1 ? `(${this.taskRun.map_index})` : '')
+        }
+      }
     }
   },
   watch: {},
   methods: {},
   apollo: {
-    flowRun: {
-      query: require('@/graphql/Schematics/flow-run.gql'),
-      variables() {
-        return {
-          id: this.flowRunId
-        }
+    taskRunDependencies: {
+      query() {
+        return this.query
       },
       skip() {
         return (
-          !this.flowRunId ||
-          (this.tasks?.length && this.flowRun.state !== 'Running')
-          // Had to add the second portion here because this
-          // query was running out of control for some reason
+          !this.taskRun?.task?.downstream_edges?.length &&
+          !this.taskRun?.task?.upstream_edges?.length
         )
       },
       update(data) {
-        if (data.flow_run && data.flow_run.length) {
-          let taskRuns = data.flow_run[0].task_runs.filter(tr =>
-            this.taskIds.includes(tr.task.id)
+        const dependencies = Object.keys(data).map(key => {
+          const task = data[key][0]
+
+          // We filter these because the references don't exist on this subset of tasks
+          task.task.upstream_edges = task.task.upstream_edges.filter(
+            edge => this.taskRun.task.id == edge.upstream_task.id
           )
 
-          taskRuns.forEach(tr => {
-            tr.flow_run_name = data.flow_run[0].name
-            tr.task.upstream_edges = tr.task.upstream_edges.filter(edge =>
-              this.taskIds.includes(edge.upstream_task.id)
-            )
-            tr.task.downstream_edges = tr.task.downstream_edges.filter(edge =>
-              this.taskIds.includes(edge.downstream_task.id)
-            )
-          })
-
-          this.tasks = taskRuns
-          return data.flow_run[0]
-        } else {
-          this.tasks = []
-        }
+          task.task.downstream_edges = task.task.downstream_edges.filter(
+            edge => this.taskRun.task.id == edge.downstream_task.id
+          )
+          return task
+        })
+
+        dependencies.forEach(d => {
+          d.flow_run_name = this.taskRun.flow_run.name
+
+          d.task.name =
```

Comment:
Is it possible that we're adding the map index twice? (Here and also in the schematic flow?). 
<img width="713" alt="Screen Shot 2020-10-22 at 11 03 43 AM" src="https://user-images.githubusercontent.com/40272060/96891127-56be7f00-1456-11eb-8731-1c507424a092.png">

Reply:
Yup, nice catch!